### PR TITLE
[Monitoring]Validate namespace in workload metrics query

### PR DESCRIPTION
**Pull request by orangedeng**
**Problem:**
It can fetch workload metrics from other project.

**Solution:**
Should validate the workload if it is in this project or not.

### CODE REVIEW

**Comment by alena1108** [comment]
@orangedeng can we link it to the corresponding github ticket

File: pkg/api/customization/monitor/handler_utils.go [hunk 1]
```diff
@@ -223,6 +224,9 @@ func parseMetricParams(userContext *config.UserContext, nodeLister v3.NodeLister
 		if err != nil {
 			return newMetricParams, err
 		}
+		if !validateNS(newMetricParams, ns) {
+			return nil, httperror.NewAPIError(httperror.PermissionDenied, "can't access to namespace "+ns+" in this project")
```
Comment (alena1108) [inline pkg/api/customization/monitor/handler_utils.go]: Shall we log the project name as well? We should also alter the error message slightly. "Can't access namespace <name> from project <>"

File: pkg/api/customization/monitor/handler_utils.go [hunk 2]
```diff
@@ -266,6 +270,9 @@ func parseMetricParams(userContext *config.UserContext, nodeLister v3.NodeLister
 			return nil, fmt.Errorf("pod name is empty")
 		}
 		ns, name := ref.Parse(podName)
+		if !validateNS(newMetricParams, ns) {
```
Comment (alena1108) [inline pkg/api/customization/monitor/handler_utils.go]: same here, need to change an error message

**Comment by orangedeng** [comment]
Updated

**Comment by aiwantaozi** [comment]
LGTM

**Comment by cjellick** [comment]
lgtm, ill let @alena1108 have one more look before we decide to merge

**Comment by alena1108** [comment]
LGTM

**Pull request merged**